File: 07-stacks-queues/07_rpn.c
#include "../test_runner.h"

int eval_rpn(char** tokens, int n) {
    return 0;
}

int main() {
    char* t1[] = {"2","1","+","3","*"};
    TEST_INT("basic", eval_rpn(t1, 5), 9);
    char* t2[] = {"4","13","5","/","+"};
    TEST_INT("division", eval_rpn(t2, 5), 6);
    TEST_SUMMARY();
}

File: 07-stacks-queues/07_rpn.py
import sys, os; sys.path.append(os.path.join(os.path.dirname(os.path.abspath(__file__)), '..'))
from test_runner import run_tests


# Evaluate Reverse Polish Notation (LC #150) -- Medium
# Evaluate postfix expression.
#
#   Key: Push numbers. On operator, pop two, compute, push result.

def eval_rpn(tokens):
    pass


run_tests(eval_rpn, [
    ((["2","1","+","3","*"],),           9),
    ((["4","13","5","/","+"],),          6),
    ((["10","6","9","3","+","-11","*","/","*","17","+","5","+"],), 22),
])
